### ISSUE TITLE
Application fist launch need the Network permission

### ISSUE THREAD

**Issue by z234009184** [ivanvorobei/SPPermissions#282]
**Feature Description**
Describe what functionality you want to see.

Hi,

    Thank you for your development of this library. I have a small demand. When the application is started for the first time, it mistakenly touches that it is not allowed to use the network permission. How can I know that the user does not open the network permission and let the user go to the setting page to open the network permission?

```
let data = CTCellularData()
        let state = data.restrictedState
        switch state {
          case .notRestricted:
          DispatchQueue.main.async {
              authorizedHandler()
          }
          default:
          DispatchQueue.main.async {
              unAuthorizedHandler()
          }
        }
```

**Comment by ivanvorobei** [comment]
Hi! 
Which permission do you using for request?

**Comment by z234009184** [comment]
![image](https://user-images.githubusercontent.com/12725906/156133617-341564ff-225a-40d3-926a-8814f9c6e343.png)

Now SPPermission can not support Network permission
 Please help

**Comment by ivanvorobei** [comment]
So you want to add network permission, right?

**Comment by z234009184** [comment]
I'd appreciate it if you could

**Comment by yuchenz27** [comment]
I am also very interested in the wireless data permission, I guess that is what you called the network permission right?

**Comment by z234009184** [comment]
Yes，in China, when the application is started for the first time, it will apply to the system for network permission. If the user accidentally prohibits the network permission, the application will not be able to access the network until the user goes to the setting center to open the wireless network permission.

**Comment by ivanvorobei** [comment]
@z234009184 fully right, no way to simple manage this permission. I can to get state of permission and request it once. 
But sorry, busy with my work now.

**Comment by ivanvorobei** [comment]
@z234009184 are you still need this permission?

**Comment by z234009184** [comment]
> @z234009184 are you still need this permission?

Yes, I think it is necessary to add network permissions as an extension, so that you can only use the PermissionsKit

**Comment by ivanvorobei** [comment]
Ok, doing it

**Comment by ivanvorobei** [comment]
I did some refresh. Available one way to request permission, but no way to get status correctly. Each time need to do request permission for get state. 
So apple hide API and its doesn't look like solution. Any ideas?

**Comment by yuchenz27** [comment]
Hey I am coming from #280. I think the permission discussed in this issue is Wireless Data Perission not Local Network Permissions.

I am looking for solutions for both of these permissions.

**Comment by ivanvorobei** [comment]
@yuchenz27 I think you right, my bad. If you have code share with me, I am ready to integrate it shortly

**Comment by ivanvorobei** [comment]
@yuchenz27 thanks for your research! I will think how I can integrate it

**Comment by yuchenz27** [comment]
@ivanvorobei no problem, your plugin is awesome. I don't think those solutions are perfect, probably there is a better way.

**Comment by yuchenz27** [comment]
Just a reminder, the code I left for Wireless Data Permission is not actually for the permission. It just checks if there is a valid network connection. So if you grant the permission and close the cellular data on your iPhone, it will still return false.
